lib/auth/utils.coffee

```diff
@@ -62,7 +62,9 @@ exports.isTokenValid = (sessionToken) ->
 	if not sessionToken? or _.isEmpty(sessionToken.trim())
 		return Promise.resolve(false)
 
-	return resin.token.get().then (currentToken) ->
+	return resin.auth.getToken()
+	.catchReturn(undefined)
+	.then (currentToken) ->
 		resin.auth.loginWithToken(sessionToken)
 			.return(sessionToken)
 			.then(resin.auth.isLoggedIn)
```

Comment:
Yeah, this method has been around for a long time, and imo it's pretty confusing, but I'm not super keen to rip it out and replace right now. I've renamed it and updated all the docs to make it clearer though, because you're totally right that this makes no sense.